python/pyarrow/io.pxi

```diff
@@ -1311,7 +1311,10 @@ cdef class Buffer(_Weakrefable):
         -------
         : bytes
         """
-        return self.buffer.get().ToHexString()
+        if self.is_cpu:
+            return self.buffer.get().ToHexString()
+        else:
+            raise NotImplementedError("Implemented only for data on CPU device")
```

Comment:
```suggestion
        if not self.is_cpu:
            raise NotImplementedError("Implemented only for data on CPU device")
        return self.buffer.get().ToHexString()            
```

(in this case your code is also perfectly fine, but I think I have a slight preference over always using the same pattern of checking if not CPU. Although it is only two lines and we only need to do it in three places for now, it could also be factored out in a helper method like `self._assert_cpu()`)